### PR TITLE
Sample word count project in F#.

### PR DESCRIPTION
**Pull request by isaacabraham**

### CODE REVIEW

**Comment by skaarthik** [comment]
Looks like F# example is failing in AppVeyor. I do not know what is needed to build fsproj. Take a look at Build.cmd and see if there can be a quick fix.

**Comment by skaarthik** [comment]
Looks like the build failure in fsproj is in line 52 "<Import Project="$(FSharpTargetsPath)" />"

**Comment by isaacabraham** [comment]
That's right.

**Comment by forki** [comment]
yeah

**Comment by skaarthik** [comment]
Extracting mvn tar file failed when building PR in AppVeyor. I just started rebuild of PR in AppVeyor.

File: examples/Batch/WordCountFSharp/App.config [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    </appender>
+  </log4net>
+  <appSettings>
+    <!--********************************************************************************************************-->
+    <!--** Uncomment the following settings to run Spark driver executable in **local** or **debug** modes ** -->
+    <!--** In debug mode, the driver is not launched by CSharpRunner but launched from VS or command prompt not configured for SparkCLR ** -->
+    <!--** CSharpBackend should be launched in debug mode as well and the port number from that should be used below ** -->
+    <!--** Command to launch CSharpBackend in debug mode is "sparkclr-submit.cmd debug" ** -->
+    <!--********************************************************************************************************-->
+    <add key="CSharpWorkerPath" value="C:\Users\Isaac\Source\Repos\Mobius\examples\Batch\WordCount\bin\Debug\CSharpWorker.exe"/>
```
Comment (skaarthik) [inline examples/Batch/WordCountFSharp/App.config]: Keeping these two settings uncommented will break the app in the non-debug mode. Please comment them out.

File: examples/Batch/WordCountFSharp/App.config [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    <add key="CSharpBackendPortNumber" value="11097"/>
+  </appSettings>
+  <system.web>
```
Comment (skaarthik) [inline examples/Batch/WordCountFSharp/App.config]: system.web element block could be removed

File: examples/Batch/WordCountFSharp/CSharpWorker.exe.config [hunk 3]
```diff
@@ -0,0 +1,40 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
```
Comment (skaarthik) [inline examples/Batch/WordCountFSharp/CSharpWorker.exe.config]: This file should be removed from the PR. The worker.exe and its config file should be from nuget.

File: examples/Batch/WordCountFSharp/WordCountFSharp.fsproj [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
+    <DebugSymbols>true</DebugSymbols>
+    <DebugType>full</DebugType>
+    <Optimize>false</Optimize>
+    <Tailcalls>false</Tailcalls>
+    <OutputPath>bin\Debug\</OutputPath>
+    <DefineConstants>DEBUG;TRACE</DefineConstants>
+    <WarningLevel>3</WarningLevel>
+    <DocumentationFile>bin\Debug\WordCountFSharp.XML</DocumentationFile>
+    <StartArguments>file:///C:/Users/Isaac/Source/Repos/Mobius/build/runtime/data/words.txt</StartArguments>
```
Comment (skaarthik) [inline examples/Batch/WordCountFSharp/WordCountFSharp.fsproj]: You may want to delete this

**Comment by skaarthik** [comment]
Build passed. I have added some comments on the changes. Will merge the PR once they are addressed.

It would be great if you can add a note to https://github.com/Microsoft/Mobius/blob/master/notes/running-mobius-app.md#running-mobius-examples-in-local-mode on running the FSharpExample.

**Comment by skaarthik** [comment]
I have verified the experience in VS when loading the solution with F# project without F# installed. As @dsyme commented in https://github.com/Microsoft/Mobius/pull/434, VS displayed the option to install F#. Skipping that did not impact building the solution and rest of the projects were built successfully.

I also verified running WordCountFSharp.exe using sparkclr-submit.cmd and it worked fine. 

@isaacabraham Thanks for this first step towards F# in Mobius,

File: examples/Batch/WordCountFSharp/App.config (hunk 2 above)
Comment (isaacabraham) [inline examples/Batch/WordCountFSharp/App.config]: This is also in all of the other ones e.g. CSharp Word Count.

File: examples/Batch/WordCountFSharp/CSharpWorker.exe.config (hunk 3 above)
Comment (isaacabraham) [inline examples/Batch/WordCountFSharp/CSharpWorker.exe.config]: Ditto - this is also in the CSharp Word Count app?

File: examples/Batch/WordCountFSharp/WordCountFSharp.fsproj (hunk 4 above)
Comment (isaacabraham) [inline examples/Batch/WordCountFSharp/WordCountFSharp.fsproj]: What and why?

Comment (skaarthik) [inline examples/Batch/WordCountFSharp/WordCountFSharp.fsproj]: "StartArguments" element.

In C# projects this entry will be in .user file which will not get checked into repo. Looks like F# projects do not keep user specific settings in a different file.

Comment (isaacabraham) [inline examples/Batch/WordCountFSharp/WordCountFSharp.fsproj]: I'm not sure what you're referring to here. Configuration Management is not C#-specific, it's a .NET thing. So if it works in C# it should work in F#. The only user settings I'm aware of are related to custom Application Settings / Configuration. Can you give me an example of this .user file?

**Pull request merged**